### PR TITLE
added french translation

### LINKED ISSUE #296: French translation

### PR DESCRIPTION
**Pull request by tomtomcrypto**
# Fixes #296

## Description

Adds french translation

## Test scenarios

Navigate the site and verify every text is in french

### CODE REVIEW

File: src/i18n/fr-fr/index.js [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+        explore_transactions: 'Explorer les transactions',
+        recent_transactions: 'Transactions récentes',
+        telos_evm_explorer: 'Exploreur Telos EVM',
+        rpc_endpoints: 'RPC',
+        monitor: 'Moniteur',
+        oops: 'Oops. Il n\'y a rien ici...',
+        go_home: 'Accueil',
+        select_sol_file: 'Sélectionner votre fichier .sol pour l\'envoyer',
+        select_json_file: 'Select standard JSON input object file for upload',
+        paste_contract_contents: 'you must select a file for upload or toggle input to paste contract contents',
```
Comment (Viterbo) [inline src/i18n/fr-fr/index.js:120]: These two lines need some french love

File: src/i18n/fr-fr/index.js [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+        runs_value_for_optimization: 'Nombre d\'itération d\'optimisation',
+        constructor_arguments: 'Arguments du constructeur',
+        comma_seperated_values: 'valeurs séparées par une virgule ex: Bob,123,0x12345...',
+        no_trailing_commas: 'pas de virgule finale',
+        paste_contract_code_here: 'copier/coller le code du contract ici...',
+        enter_contract_text: 'Entrer ou coller le texte du contat',
+        verify_contract: 'Verifier le contrat',
+        reset: 'Réinitialiser',
+        gas_used: 'Gaz utilisé',
+        transactions: 'Transactions',
```
Comment (Viterbo) [inline src/i18n/fr-fr/index.js:140]: Is this word the same as in English?

File: src/i18n/fr-fr/index.js [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+        erc20_transfers: 'Transferts ERC20',
+        erc721_transfers: 'Transferts ERC721',
+        erc1155_transfers: 'Transferts ERC1155',
+        tokens: 'Jetons',
+        created_at_trx: 'Cré par',
+        by_address: 'Par Adresse',
+        number_used_once: 'Nombre utilisé une fois (nonce)',
+        native_account: 'Compte natif',
+        balance: 'Solde',
+        view_source_prompt: 'This contract has been verified. You can view the source code & metadata in the \'contract\' tab',
```
Comment (Viterbo) [inline src/i18n/fr-fr/index.js:152]: can you do the same here?

File: src/i18n/fr-fr/index.js [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+        account_url: '{ domain }/account/{ account }',
+        tlos_balance: '{ balance } TLOS',
+        couldnt_retreive_metadata_for_address: 'Could not retreive metadata for { address }: { message }',
```
Comment (Viterbo) [inline src/i18n/fr-fr/index.js:155]: also this one

File: src/i18n/fr-fr/index.js [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+        gas_limit: 'Limite de gaz',
+        nonce: 'Nonce',
+        input: 'Entrée',
+        output: 'Sortie',
+        value: 'Valeur',
+    },
+    components: {
+        internal_txns: 'Internal Transactions',
+        n_internal_txns: '{ amount } internal transactions',
+        none: 'None',
```
Comment (Viterbo) [inline src/i18n/fr-fr/index.js:189]: These three lines are new, please make them feel comfortable

File: src/i18n/fr-fr/index.js [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+            readonly: 'Ce champ est en lecture seule',
+            required: 'Ce champ est requis',
+            too_large: 'La valeur maximum pour int{ size } est { max }',
+            too_small: 'La valeur minimum pour int{ size } est { min }',
+            too_large_pow2: 'La valeur maximum pour int{ size } est 2^{ size } - 1',
+            too_small_pow2: 'La valeur minimum pour int{ size } est -(2^{ size }) + 1',
+            too_large_unsigint: 'La valeur maximum pour uint{ size } est 2^{ size } - 1',
+            no_negative_unsigint: 'La valeur pour uint{ size } ne peut pas être négative',
+            invalid_signed_integer: 'Entier signé non valide',
+            invalid_unsigint: 'Entry must be a valid unsigned integer',
```
Comment (Viterbo) [inline src/i18n/fr-fr/index.js:286]: almost forgotten...

File: src/i18n/fr-fr/index.js [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+            checked_at: 'Checked at',
+            task: 'Tâche',
+            message: 'Message',
+            block_height: 'Bloc',
+            latency: 'Latence',
+            click_to_change_format: 'Cliquer pour changer de format',
+        },
+        contract_tab: {
+            copy_abi_to_clipboard: 'Copy contract ABI JSON to clipboard',
+            enter_amount: 'Select number of decimals and enter an amount, this will be entered for you into the function parameter as uint256',
```
Comment (Viterbo) [inline src/i18n/fr-fr/index.js:304]: These two lines need some love too.

File: src/i18n/fr-fr/index.js [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+            read: 'Lire',
+            write: 'Ecrire',
+            amount: 'Montant',
+            value: 'Valeur',
+            custom_decimals: 'Decimales personalisés',
+            custom: 'Personalisé',
+            unverified_contract_source: 'Ce contrat n\'a pas été vérifié.',
+            click_here: 'Cliquez ici',
+            upload_source_files: 'to upload source files and verify this contract. ' +
+                'Alternatively, you can interact with the contract using an arbitrary ABI:',
```
Comment (Viterbo) [inline src/i18n/fr-fr/index.js:317]: in between the lines...

File: src/i18n/fr-fr/index.js [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+            read_functions: 'Fonctions de lecture',
+            write_functions: 'Fonctions de d\'écriture',
+            unverified_contract: 'Contract non verifié',
+        },
+        header: {
+            sign_in: 'Connexion',
+            sign_out: 'Déconnexion',
+            liq_staking: 'Staking Liquide',
+            goto_staking: 'Aller au Staking',
+            goto_health_monitor: 'go to Heath Monitor page',
```
Comment (Viterbo) [inline src/i18n/fr-fr/index.js:334]: last one I found... for now... hahahah...

**Review by Viterbo — CHANGES_REQUESTED** [review]
@tomtomcrypto, I updated the file with [this PR 128fef7](https://github.com/telosnetwork/teloscan/pull/349/commits/128fef714006f26388d1e576f248134a0ba915f4) adding three more lines.

I also take the time to comment on every English line I found that may help you to identify them.

Thank you!

File: src/i18n/fr-fr/index.js (hunk 2 above)
Comment (tomtomcrypto) [reply src/i18n/fr-fr/index.js:140]: yep

File: src/i18n/fr-fr/index.js (hunk 9 above)
Comment (tomtomcrypto) [reply src/i18n/fr-fr/index.js:334]: It is still a work in progress, I haven't translated everything yet, just doing it when I got downtime

**Comment by tomtomcrypto** [comment]
Doing a final reading and a few trans missing and we'll be good

**Review by karynemayer — APPROVED** [review]

**Review by donnyquixotic — APPROVED** [review]

**Pull request merged**